filesystem/trd.ksy

```diff
@@ -0,0 +1,138 @@
+meta:
+  id: trd
+  file-extension: trd
+  title: "TR-DOS flat-file disk image"
+  license: CC0-1.0
+  # It's extended ascii (with ZX Spectrum Basic tokens, UDG, etc), but
+  # "unknown-8bit" can't be used for some reason
+  encoding: ascii
```

Comment:
This is obviously borderline territory, but have you checked if anyone actually has name for this encoding (i.e. IANA, IBM, Java encodings registry, etc?)

Leaving it as ASCII would obviously break stuff if a non-ASCII char will be encountered. Using no encoding and treating these strings as byte arrays would be ugly in terms of representation.

Reply:
No, there's no such encoding listed everywhere. Most of [extended characters](https://en.wikipedia.org/wiki/ZX_Spectrum_character_set) are not representable as Unicode code points. AFAIK, in languages with checked unicode strings such as Rust such strings can't be represented as string type. So I think the only way to safely handle this is to make filenames byte arrays, then code that uses decoder generated from this definition can convert (with loss) to string representation if needed. Replacing Basic tokens, etc, is outside of scope of filesystem definition, I think.

However, Kaitai IDE will not show decoded filenames in "object tree" in this case.

IANA character set registry has `UNKNOWN-8BIT` charset, but it causes exception in Kaitai IDE, and AFAIK, it has the same semantics as treating string as byte array (it's not real encoding and not representable in unicode).

Reply:
I changed all strings (file names, disk label, disk password) to byte arrays. However, I had to add indirection to file name (it's now available as `name.name`) because of limitation that byte arrays don't have `_io` ([this was also in recent discussion on gitter](https://gitter.im/kaitai_struct/Lobby?at=5be02324cd2a6b5bdc9b89f9)), and I can't find another way to extract first byte from byte array.

Reply:
Thanks for the explanation. Indeed, this looks pretty much like situation with "ANSI" or "OEM" character set, which essentially means that exact set of characters displayed is unknown and depends on system's settings. Indeed, typically we treat that as encoding-less byte array.

You're totally right, byte arrays nowadays lack `a[x]` operator, and I guess that's something that we should address.